### PR TITLE
Updating for Chef 12.4.x, fixes #240

### LINKED ISSUE #240: Specs fail with stack overflow message
Many (1/3?) of the specs in master (36b25276d627e5af6067d2293c32adc20e312091, Ruby 2.1.2) are failing with this:

  1) Chef::Resource::AwsEbsVolume When the Chef server exists when the chef repo exists and when connected to AWS aws_ebs_volume 'test_volume' creates an ebs volume
     Failure/Error: Unable to find matching line from backtrace
     SystemStackError:
       stack level too deep

I wasn't paying attention to my Chef version, but this corresponds roughly with the realize of Chef 12.4.

### PR DESCRIPTION
**Pull request by tyler-ball**
Fixes #240 

It broke some resource hacks we were doing.  I ran the tests and they pass with both 12.3 and 12.4

\cc @fnichol @randomcamel

### CODE REVIEW

File: spec/integration/aws_key_pair_spec.rb [hunk 1]
```diff
@@ -10,11 +10,11 @@
       end
 
       it "aws_key_pair 'test_key_pair' creates a key pair" do
-        expect_recipe {
+        expect(recipe {
```
Comment (randomcamel) [inline spec/integration/aws_key_pair_spec.rb]: Wait, `recipe {}` converges the recipe?

**Comment by randomcamel** [comment]
One question, not a blocker. :+1:

**Comment by fnichol** [comment]
Looking decent to me as well, :+1:

File: spec/integration/aws_key_pair_spec.rb (hunk 1 above)
Comment (tyler-ball) [inline spec/integration/aws_key_pair_spec.rb]: Exactly what I was going to link you!

**Pull request merged**